clients/views.py

```diff
@@ -59,16 +53,19 @@ def get(self, request, *args, **kwargs):
 
 
 
-class ClientViewView(LoginRequiredMixin,TemplateView):
+class ClientView(LoginRequiredMixin,TemplateView):
 	template_name = 'clients/view_client.html'
 	
 	def get(self, *args, **kwargs):
-		context = {
+		client = Client.objects.get(id=kwargs['client_id'])
+		if self.request.user == client.owner: 
+			context = {
 				'client': get_object_or_404(Client, id=kwargs['client_id']),
 				'current_path': self.request.path,
 				'invoices' : Invoice.objects.filter(client=kwargs['client_id']),
```

Comment:
nag query ka ug client object then here nag query nasad ka ug invoice. for the same data.

your total queries: 3

whereas you can get all those data in just 1 query.

analyse your code and what you want to do. DILI PURKIT MUGANA KAY PWEDE NA